Prox/Prox/Data/PlacesController.swift

```diff
@@ -110,13 +117,36 @@ class PlacesProvider {
         }
     }
 
-    private func displayPlaces(places: [Place], forLocation location: CLLocation) {
-        let preparedPlaces = self.preparePlaces(places: places, forLocation: location)
+    private func fetchPlacesWithEvents(location: CLLocation) {
+        let eventProvider = EventsProvider()
+        eventsPlaces.removeValue(forKey: location)
+        eventProvider.getEventsWithPlaces(forLocation: location, usingPlacesDatabase: database) { places in
+            self.eventsPlaces[location] = places
+            self.displayPlaces(forLocation: location)
+        }
+    }
+
+    private func displayPlaces(forLocation location: CLLocation) {
+        guard let places = self.nearbyPlaces[location],
+            let eventPlaces = self.eventsPlaces[location] else { return }
```

Comment:
`displayPlaces` will get called multiple times for `nearbyPlaces` (each time around the loop while we poll to see if the database has been populated yet) and once for `eventPlaces`. The idea had been that we only tell the delegate once that there are new places once we have them all, however I have now updated it so that we tell the delegate every time `displayPlaces` is called, but only merge the results once we have both. Once we have both, we clear the cache and start from fresh.

These two calls currently run concurrently and I would rather that than having them run in sequence seeing as it can take a very long time for the places fetch to return.

I've made this clearer